otter/convergence/service.py

```diff
@@ -531,6 +539,20 @@ def converge_one_group(currently_converging, recently_converged,
         # unexpected errors, so convergence will be retried.
         yield err(None, 'converge-non-fatal-error')
     else:
+        if result[0] is StepResult.LIMITED_RETRY:
+            # We allow further iterations to proceed as long as we haven't been
+            # waiting for a LIMITED_RETRY for N consecutive iterations.
+            if (yield waiting.read())[group_id] > limited_retry_iterations:
```

Comment:
Correct me if I am wrong but wouldn't it raise `KeyError` when group returns LIMITED_RETRY the first time? Maybe you need `defaultdict` here?